### ISSUE TITLE
label length

### ISSUE THREAD

**Issue by samueldmcdermott** [SebastianBocquet/pygtc#34]
I'm getting an `AttributeError`.  The traceback is
```
File ~/micromamba/envs/e2/lib/python3.11/site-packages/pygtc/pygtc.py:935, in plotGTC(chains, **kwargs)
    932 # x labels
    933 # Get label length of the bottom row
    934 for i in range(len(axH)):
--> 935     bboxTickLabel = (axH[i].xaxis.get_ticklabel_extents(renderer)[0]
    936                      .get_points())
    937     bboxSize[i] = bboxTickLabel[1, 1]-bboxTickLabel[0, 1]
    938     panelWidth = (axH[i].get_window_extent()
    939                   .transformed(fig.dpi_scale_trans.inverted()).width)

AttributeError: 'XAxis' object has no attribute 'get_ticklabel_extents'
```

This attribute seems to have been [deprecated](https://matplotlib.org/3.7.0/api/_as_gen/matplotlib.axis.Axis.get_ticklabel_extents.html) as of `matplotlib 3.6`. Do you know a quick workaround?

**Comment by FaustinCarter** [comment]
The quick fix for the moment is to downgrade to 3.6 (or maybe you could get away with 3.7 since they usually do removals the second point release after a deprecation). 

In the meantime we will have to have a look at whether we can put some time in to update the codebase.

**Comment by samueldmcdermott** [comment]
I think I found a simple solution using `get_tightbbox`. It leads to an error for one of the y-axes for reasons I think I understand, so I also added some duck typing to get around that, and it works on a very naive implementation. It passes tests (though I get lots of warnings from `pytest` due to `return`s instead of `assert`s). Would you prefer more discussion here or should I issue a pull request?
